src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -2558,6 +2558,121 @@ void EmitSingleCharLoop(RegexNode node, RegexNode? subsequent = null, bool emitL
                 MarkLabel(endLoop);
             }
 
+            void EmitSingleCharLazy(RegexNode node, RegexNode? subsequent = null, bool emitLengthChecksIfRequired = true)
```

Comment:
I suppose there is no good way to have the generator and compiled writers to share the code?

Reply:
.... I'm sure you've already thought about this 😆

Reply:
There's nothing better today. We've talked about possible approaches, whether it be supporting emitting IL blobs as part of a C# compilation (but then you lose all the nice benefits of easy-to-read generated C#), building Roslyn into core libraries so that you emit C# and compile to IL as part of RegexOptions.Compiled, coming up with some kind of middle ground API that's capable of outputting either IL or C# for common patterns and targeting that, etc.

Reply:
> some kind of middle ground API that's capable of outputting either IL or C#

Another variation of that is perhaps -- we extract as much interesting logic from the generator/compiler and back into the shared writer code, perhaps by having higher level opcodes. The test being that optimization work rarely is in the generator/compiler and they are relatively "dumb". Of course, I haven't thought this through at all.